### PR TITLE
[Java] fix generation for JavaTimeFormatter

### PR DESCRIPTION
**Pull request by borsch**
Fix for issue: https://github.com/OpenAPITools/openapi-generator/issues/7952

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by borsch** [comment]
please review

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java [hunk 1]
```diff
@@ -550,6 +550,7 @@ public void processOpts() {
             setDateLibrary(additionalProperties.get("dateLibrary").toString());
         }
 
+        additionalProperties.put("dateLibrary_" + dateLibrary, "true");
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:553]: In the new code, if you look at line 555, 556, 562, 568, etc, we already add mustache tags for date libraries. why do we need `dateLibrary_${dateLibrary}` in partciuarl?

**Comment by borsch** [comment]
close to restart travis

**Review by wing328 — APPROVED** [review]
Tested locally and the result is good.

**Pull request merged**